homeassistant/components/command_line/sensor.py

```diff
@@ -36,6 +41,7 @@
         vol.Optional(CONF_COMMAND_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
         vol.Optional(CONF_JSON_ATTRIBUTES): cv.ensure_list_csv,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
```

Comment:
So there is an issue with this, this PR violates ADR-0007 at this point.

We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Reply:
So this would take it from a minor change to a breaking change.  I feel like it would be best to break it all the way and change it so that the configuration would be done with the GUI.  It would be a lot more work, and there are still plenty of things that I don't have a full grasp on for making that large of a change, but it would save from having two breaking changes down the line.  

Is there a preference for which route to take?

Reply:
See #63840

Reply:
It looks like #63840 would solve the ADR-0007 issue and the code appears mostly compatible.  The main change is the `def setup_platform` --> `async def async_setup_platform` section.